app/enrichment/create-invoice-number.js

```diff
@@ -1,11 +1,28 @@
 const createInvoiceNumber = (paymentRequest) => {
   try {
-    return `S${paymentRequest.invoiceNumber
-    ? paymentRequest.invoiceNumber.slice(-7)
-    : paymentRequest.agreementNumber.slice(-7)}${paymentRequest.contractNumber}V${paymentRequest.paymentRequestNumber.toString().padStart(3, '0')}`
+    switch (paymentRequest.schemeId) {
+      case 1:
+      case 2:
+      case 3:
+        return createSitiAgriInvoiceNumber(paymentRequest)
+      default:
+        return createDefaultInvoiceNumber(paymentRequest)
+    }
   } catch {
     return undefined
   }
 }
 
+const createSitiAgriInvoiceNumber = (paymentRequest) => {
+  if (paymentRequest.invoiceNumber.length >= 7 && paymentRequest.contractNumber && paymentRequest.paymentRequestNumber) {
```

Comment:
7 here is like magic number, consider something like const minInvoiceLength

Reply:
Yes this is a fair point.

I've named the variable `sitiInvoiceNumberElementLength` as this value relates to the number element of the original invoice number.